web/src/main/java/se/inera/intyg/webcert/web/csintegration/certificate/CertificateStatusUpdateFactory.java

```diff
@@ -0,0 +1,102 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package se.inera.intyg.webcert.web.csintegration.certificate;
+
+import static se.inera.intyg.common.support.Constants.HSA_ID_OID;
+
+import java.io.ByteArrayOutputStream;
+import java.nio.charset.StandardCharsets;
+import java.time.LocalDateTime;
+import java.util.Base64;
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBElement;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Marshaller;
+import org.w3._2002._06.xmldsig_filter2.XPathType;
+import se.inera.intyg.common.support.common.enumerations.HandelsekodEnum;
+import se.inera.intyg.common.support.facade.model.Certificate;
+import se.inera.intyg.common.support.modules.support.api.notification.ArendeCount;
+import se.inera.intyg.common.support.xml.XmlMarshallerHelper;
+import se.inera.intyg.webcert.notification_sender.notifications.services.NotificationTypeConverter;
+import se.inera.intyg.webcert.notification_sender.notifications.util.NotificationRedeliveryUtil;
+import se.riv.clinicalprocess.healthcond.certificate.certificatestatusupdateforcareresponder.v3.CertificateStatusUpdateForCareType;
+import se.riv.clinicalprocess.healthcond.certificate.certificatestatusupdateforcareresponder.v3.ObjectFactory;
+import se.riv.clinicalprocess.healthcond.certificate.registerCertificate.v3.RegisterCertificateType;
+import se.riv.clinicalprocess.healthcond.certificate.types.v3.DatePeriodType;
+import se.riv.clinicalprocess.healthcond.certificate.types.v3.HsaId;
+import se.riv.clinicalprocess.healthcond.certificate.types.v3.PQType;
+import se.riv.clinicalprocess.healthcond.certificate.types.v3.PartialDateType;
+
+public class CertificateStatusUpdateFactory {
+
+    private CertificateStatusUpdateFactory() {
+        throw new IllegalStateException("Utility class");
+    }
+
+    public static byte[] create(Certificate certificate, String encodedXmlRepresentation, HandelsekodEnum eventType,
+        LocalDateTime now) {
+        final var request = getRegisterCertificateType(encodedXmlRepresentation);
+        
+        final var certificateStatusUpdateForCareType = new CertificateStatusUpdateForCareType();
+        certificateStatusUpdateForCareType.setIntyg(request.getIntyg());
+        certificateStatusUpdateForCareType.setHandelse(
+            NotificationRedeliveryUtil.getEventV3(eventType, now, null, null)
+        );
+        certificateStatusUpdateForCareType.setSkickadeFragor(NotificationTypeConverter.toArenden(ArendeCount.getEmpty()));
+        certificateStatusUpdateForCareType.setMottagnaFragor(NotificationTypeConverter.toArenden(ArendeCount.getEmpty()));
+        certificateStatusUpdateForCareType.setHanteratAv(
+            NotificationRedeliveryUtil.getIIType(new HsaId(), certificate.getMetadata().getIssuedBy().getPersonId(), HSA_ID_OID)
```

Comment:
I think we have to get the hanteratAv based on who is logged in, except for the situation when the certificate draft is created.

Reply:
Updated so we either take the hsaId from logged in WebcertUser or IntygUser.